leftwm-core/src/utils/child_process.rs

```diff
@@ -87,6 +88,31 @@ impl Nanny {
     /// Could be caused by inadequate permissions.
     pub fn run_global_up_script() -> Result<Option<Child>> {
         let mut path = Self::get_config_dir()?;
+        let config_dir = fs::read_dir(&path)?;
+
+        let mut scripts = std::collections::BinaryHeap::new();
+
+        for entry in config_dir {
+            let entry = match entry {
+                Ok(e) => e,
+                Err(_) => continue,
+            };
```

Comment:
I'd just use `if let Some(entry) = entry { ... }` also check if entry is a file before collecting it!

Reply:
> I'd just use `if let Some(entry) = entry { ... }`

Right!

`Self::run_script` already check if the argument is a file before executing. Should we check two times?

Reply:
> Self::run_script already check if the argument is a file before executing. Should we check two times?
 
Ok eh... this is odd... normally I'd expect the function to run immediately and return an error if it can't run (for example if it got a path to a directory instead of a file). But I think this can be fixed in another PR.

Reply:
> Ok eh... this is odd... normally I'd expect the function to run immediately and return an error if it can't run (for example if it got a path to a directory instead of a file). But I think this can be fixed in another PR.

Yeah, I agree. Maybe I could code this to just return `up` scripts in an arbitary directory.

Reply:
> Yeah, I agree. Maybe I could code this to just return up scripts in an arbitary directory.

I'd prefer this IMO, but maybe someone else has  a different opinion on this @VuiMuich what do you think?

Reply:
> I'd prefer this IMO, but maybe someone else has a different opinion on this @VuiMuich what do you think?

Right now I've coded this solution to be generic over "files" and "extensions":

```rust
    fn get_files_in_path_with_ext(
        path: impl AsRef<Path>,
        ext: impl AsRef<OsStr>,
    ) -> Result<BinaryHeap<Reverse<PathBuf>>> {
        let config_dir = fs::read_dir(&path)?;

        let mut scripts = BinaryHeap::new();

        for entry in config_dir.flatten() {
            let file = entry.path();

            if let Some(extension) = file.extension() {
                if extension == ext.as_ref() {
                    scripts.push(Reverse(file));
                }
            }
        }

        Ok(scripts)
    }
```

Reply:
>  /// This function will return an error if `stf::fs::read_dir()` returns an error,
    /// for example when:
    /// * The provided `path` doesn't exist.
    /// * The process lacks permissions to view the contents.
    /// * The `path` points at a non-directory file.

I'd sum this up to:
```rust
    /// # Errors
    ///
    /// Comes from `std::fs::read_dir()`.
```
because the rest can be read there.